### PR TITLE
Test

### PR DESCRIPTION
**Pull request by noakrams**

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -9,11 +10,16 @@ all: main
 
 # Linker
 main: main.o asm.o
-	$(CC) -o main main.o asm.o $(LFLAGS)
+		$(CC) -o main main.o asm.o $(LFLAGS)
```
Comment (IdanTo1) [inline makefile]: tab is redundant

File: makefile [hunk 2]
```diff
@@ -9,11 +10,16 @@ all: main
...
 
 # Source files
 main.o: mainAssignment0.c
-	$(CC) $(CFLAGS) -o main.o mainAssignment0.c
+		$(CC) $(CFLAGS) -o main.o mainAssignment0.c
```
Comment (IdanTo1) [inline makefile]: same

File: makefile [hunk 3]
```diff
@@ -9,11 +10,16 @@ all: main
...
 
 asm.o: asmAssignment0.s
-	$(ASM) $(ASMFLAGS) -o asm.o asmAssignment0.s
+		$(ASM) $(ASMFLAGS) -o asm.o asmAssignment0.s
+		
+.PHONY: clean
+
+clean: 
+	rm -f *.o main
```
Comment (IdanTo1) [inline makefile]: add a newline at the end of the file (:

File: asmAssignment0.s [hunk 4]
```diff
@@ -1,25 +1,30 @@
...
+        pushad                   	; push all signficant registers onto stack (backup registers values)
+        mov ecx, dword [ebp+8]		; get function argument on stack
+									; now ecx register points to the input string
+	yourCode:						; use label to build a loop for treating the input string characters
+			cmp byte[ecx], 32		; check if byte equals to ASCII's space value
+			jz updateChar			; if is, go to updating the char
+			cmp byte[ecx], 9		; check if byte equals to ASCII's TAB value
+			jnz nextChar			; if not, skip to next char
+	updateChar:						; use label to update the char to "_" if necessary
+			mov byte[ecx], 95		; change char value to "_"
```
Comment (IdanTo1) [inline asmAssignment0.s]: why decimal and not Hex?

File: makefile (hunk 3 above)
Comment (noakrams) [reply makefile]: Why do we need newline at the end of the makefile?

**Pull request merged**